cranelift/codegen/src/isa/zkasm/lower.isle

```diff
@@ -116,19 +116,11 @@
       (_ InstOutput (gen_div_by_zero y2)))
     (rv_divuw (zext x ty $I64) y2)))
 
-; (rule -1 (lower (has_type (fits_in_32 ty) (sdiv x y)))
-;   (let
-;     ((a XReg (sext x ty $I64))
-;       (b XReg (sext y ty $I64))
-;       (_ InstOutput (gen_div_overflow a b ty))
-;       (_ InstOutput (gen_div_by_zero b)))
-;     (rv_divw a b)))
-;
-; (rule (lower (has_type $I64 (sdiv x y)))
-;   (let
-;     ((_ InstOutput (gen_div_overflow x y $I64))
-;       (_ InstOutput (gen_div_by_zero y))    )
-;     (rv_div x y)))
+ (rule -1 (lower (has_type (fits_in_32 ty) (sdiv x y)))
```

Comment:
You can drop this rule entirely I believe, and have something like

```
(rule (lower (sdiv x y))
     (zk_div x y))
```

Reply:
Yes, it works, added [here](https://github.com/near/wasmtime/pull/54/commits/494aefb12aeb6b99c26ce5fb92a1ca7d45bfd6fc). But maybe division to two lowering patterns will help in future when we will adding smaller types?

Reply:
I don’t think it is worth thinking about it right now – ISLE is reasonably easy to extend when needed, but quite hard to simplify later in my experience, so its better to do the simple thing first.